webui/react/src/services/apiBuilder.ts

```diff
@@ -36,16 +36,17 @@ export function generateApi<Input, Output>(api: Api<Input, Output>) {
 
       return api.postProcess ? api.postProcess(response) : response.data as Output;
     } catch (e) {
+      if (axios.isCancel(e)) return Promise.reject(e);
```

Comment:
Do we still want to reject the promise here if we're not going to handle the error?

Reply:
we're not going to handle it in a "generic" way through error handler but it's still the only way the caller would know that the request didn't succeed. (we could probably also rethrow or only conditionally call into handleError. )

Reply:
suggestion: for our purpose, `Promise.reject(e)` and `throw e` are essentially the same as we discourage API usage outside of `try/catch` blocks.
```
      if (!axios.isCancel(e)) {
        const isAuthError = isAuthFailure(e);
        handleError({
          error: e,
          level: isAuthError ? ErrorLevel.Fatal : ErrorLevel.Error,
          message: isAuthError ?
            `unauthenticated request ${api.name}` : `request ${api.name} failed.`,
          silent: true,
          type: ErrorType.Auth,
        });
      }
      throw e;
```